### ISSUE TITLE
Continuing training with niter_decay

### ISSUE THREAD

**Issue by ibro45** [junyanz/pytorch-CycleGAN-and-pix2pix#696]
Hello!

Since I have a limited uptime for GPU, I'm training my model in sequences of 15 epochs, each time saving the weights and loading them and continuing the training. I have trained 100 epochs with fixed learning rate (niter), but now I would like to be able to separate the training of the next 100 epochs with decaying learning rate (niter_decay). 

If I'm not wrong, there isn't a way for me to define that I would like it to decay through 100 epochs while doing it in series of 15 epochs. From what I understood, if I run it for the first 15 epochs (out of 100), it'll drive the learning rate to 0 by the end of that 15th epoch.

Any help is appreciated!

**Comment by junyanz** [comment]
It should be fine as long as you also set the `--epoch_count` correctly. See [here](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/master/options/train_options.py#L27) for more details. The learning decay will consider the `epoch_count`. See this [line](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/blob/41931e25c7d12e0ff2fcea4ee7ba2e597769e6f2/models/networks.py#L53).

**Comment by ibro45** [comment]
When I was training GAN without learning rate decay, I used to define it in this way:

```
checkpoint_name = 'checkpoint'
num_epoch_to_do = 15
# The total number of epochs done in previous sessions - 45 means 3 sessions x 15 epochs
previous_num_of_epochs = 45  

os.putenv("CHECKPOINT", str(checkpoint_name))
os.putenv("EPOCH_COUNT", str(previous_num_of_epochs + 1))
os.putenv("NITER", str(previous_num_of_epochs + num_epoch_to_do))
```
These environment variables would then be passed as arguments when calling the `train.py`

Unfortunately, I don't think it's possible to train with learning decay in such a fashion and I had to find another way. I eventually came up with the idea of using timeout of subprocess. The solution looks like this:
```
import subprocess

cmd = 'python train.py \
       --niter $NITER --niter_decay $NITERDECAY --dataset_mode aligned \
       --dataroot ../dataset/ --model pix2pix \
       --name $CHECKPOINT --continue_train --epoch_count $EPOCH_COUNT > ../out.txt'

seconds = 30000 # ~8,5 hours, 9 hours is kaggle's limit

try:
    subprocess.run(cmd, shell=True, timeout=360)
except subprocess.TimeoutExpired:
    print('Process stopped before the time limit')
```

The process will be run and it's output will be saved to out.txt. Once the process is terminated, I check what was the last epoch to be ran completely before the termination and use it to define the `epoch_count` of the next session.
